### PR TITLE
Add the ability to have Section Summaries

### PR DESCRIPTION
**Pull request by MIslam92**
### What is the context of this PR?
Introducing Section Summaries into Runner.
Card: https://trello.com/c/AC1836ru/1800-section-summary-m

### How to review 
Review this first: https://github.com/ONSdigital/eq-schema-validator/pull/31
Open `test_section_summary.json`, check that section summaries are working and matches design on card.
Ensure that all tests pass as usual.

### CODE REVIEW

**Review by yoganandkunche — COMMENTED** [review]
Can we only display `House Details` section only after completing `property details` section as displaying question `House details` is dependent on options selected under property details.

File: app/templates/partials/summary/sectionsummary.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{% set not_answered = 'No answer provided' %}
```
Comment (jonnyshaw89) [inline app/templates/partials/summary/sectionsummary.html]: Why cant we just use the summary.html?

File: app/templating/summary_context.py [hunk 2]
```diff
@@ -21,3 +23,17 @@ def build_summary_rendering_context(schema, schema_json, answer_store, metadata)
             groups.extend([Group(group, path, answer_store, metadata, url_for)])
 
     return groups
+
+
+def build_section_summary_rendering_context(schema, schema_json, answer_store, metadata, current_location):
```
Comment (jonnyshaw89) [inline app/templating/summary_context.py]: this seems very common with `build_summary_rendering_context` consider refactoring

File: app/views/questionnaire.py [hunk 3]
```diff
@@ -644,14 +644,28 @@ def _get_context(full_routing_path, block, current_location, form):
     if block['type'] == 'Summary':
         rendered_schema_json = renderer.render(g.schema.json, **schema_context)
         form = form or FlaskForm()
-        summary_rendered_context = build_summary_rendering_context(g.schema, rendered_schema_json, answer_store, metadata)
+        summary_rendered_context = build_summary_rendering_context(g.schema, rendered_schema_json, answer_store,
+                                                                   metadata)
```
Comment (jonnyshaw89) [inline app/views/questionnaire.py]: why?

File: app/views/questionnaire.py [hunk 4]
```diff
@@ -644,14 +644,28 @@ def _get_context(full_routing_path, block, current_location, form):
...
         context = {'form': form,
                    'summary': summary_rendered_context,
                    'variables': variables,
                    'is_view_submission_response_enabled': is_view_submitted_response_enabled(g.schema.json),
                    'answers_are_editable': True}
         return context
 
+    elif block['type'] == 'SectionSummary':
```
Comment (jonnyshaw89) [inline app/views/questionnaire.py]: this is also very common. can you not reuse and then just have the logic around the line that builds the context?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]
I think you should review https://github.com/ONSdigital/eq-survey-runner/pull/1473 before progressing with any refactoring

File: app/views/questionnaire.py (hunk 3 above)
Comment (MIslam92) [reply app/views/questionnaire.py]: PEP8 error complaining about line length

File: app/templates/partials/summary/sectionsummary.html (hunk 1 above)
Comment (MIslam92) [reply app/templates/partials/summary/sectionsummary.html]: We would have to change the design to match the card and that summary.html is used elsewhere so best to make a separate file for section summaries

File: app/views/questionnaire.py (hunk 3 above)
Comment (jonnyshaw89) [reply app/views/questionnaire.py]: but the length hasn't changed so why has this started erroring?

File: app/templates/partials/summary/sectionsummary.html (hunk 1 above)
Comment (jonnyshaw89) [reply app/templates/partials/summary/sectionsummary.html]: I don't think its best to duplicate code.
what is the difference in the 2?

File: app/views/questionnaire.py (hunk 3 above)
Comment (MIslam92) [reply app/views/questionnaire.py]: Oh no its always been there just no ones bothered to move it to a new line, similar instances of this are scattered throughout the code base as well

File: app/templates/partials/summary/sectionsummary.html (hunk 1 above)
Comment (MIslam92) [reply app/templates/partials/summary/sectionsummary.html]: Mainly title for the group being passed and text on the change answer link?

**Comment by shahi645** [comment]
Needs to be updated with master, the questionnaire_schema has changed alot

File: app/views/questionnaire.py (hunk 3 above)
Comment (shahi645) [reply app/views/questionnaire.py]: Its because of the indentation previously, needs to be within the parentheses on next line

File: app/templates/partials/summary/sectionsummary.html (hunk 1 above)
Comment (hamishtaplin) [reply app/templates/partials/summary/sectionsummary.html]: I agree with @jonnyshaw89 — don't duplicate templates

File: app/templating/view_context.py [hunk 5]
```diff
@@ -85,10 +86,22 @@ def build_view_context_for_question(current_location, variables, rendered_block,
     return context
 
 
-def build_view_context_for_summary(metadata, answer_store, schema_context, variables, csrf_token):
-    rendered_schema_json = renderer.render(g.schema.json, **schema_context)
-    summary_rendered_context = build_summary_rendering_context(g.schema, rendered_schema_json['sections'],
-                                                               answer_store, metadata)
+def build_view_context_for_summary(metadata, answer_store, schema_context, block, variables, csrf_token,
```
Comment (jonnyshaw89) [inline app/templating/view_context.py]: You are not passing in the `block` you are passing in the `block_type`

File: app/templating/view_context.py [hunk 6]
```diff
@@ -85,10 +86,22 @@ def build_view_context_for_question(current_location, variables, rendered_block,
...
+
+    if block == 'Summary':
+        summary_rendered_context = build_summary_rendering_context(g.schema, sections,
+                                                                   answer_store, metadata)
+    elif block == 'SectionSummary':
+        for section in sections:
+            group_ids = {group['id'] for group in section['groups']}
+            if current_location.group_id in group_ids:
+                section = [section]
+                summary_rendered_context = build_summary_rendering_context(g.schema, section, answer_store,
```
Comment (jonnyshaw89) [inline app/templating/view_context.py]: After this line you want to stop it from looping around the rest of the sections as you have already found your match

File: app/views/questionnaire.py [hunk 7]
```diff
@@ -673,7 +673,8 @@ def _get_front_end_navigation(answer_store, current_location, metadata, routing_
     completed_blocks = get_completed_blocks(current_user)
     navigation = Navigation(g.schema, answer_store, metadata, completed_blocks, routing_path)
     block_json = g.schema.get_block(current_location.block_id)
-    if block_json is not None and block_json['type'] in ('Question', 'Interstitial', 'Confirmation', 'Summary'):
+    if block_json is not None and block_json['type'] in ('Question', 'Interstitial', 'Confirmation', 'SectionSummary',
```
Comment (jonnyshaw89) [inline app/views/questionnaire.py]: It might be better to invert this logic now?
or even question why it is even here?

File: tests/functional/generate_pages.py [hunk 8]
```diff
@@ -194,6 +201,21 @@ def process_question(question, page_spec, num_questions, page_name):
         process_answer(question_type, answer, page_spec, long_names, page_name)
 
 
+def process_section_summary(schema_data, page_spec):
```
Comment (jonnyshaw89) [inline tests/functional/generate_pages.py]: This is almost identical to `process_summary` can they not be the same thing?

File: tests/functional/generate_pages.py [hunk 9]
```diff
@@ -281,6 +303,14 @@ def process_block(block, dir_out, schema_data, spec_file, relative_require='..')
         page_spec.write(CLASS_NAME.substitute(block_context))
         page_spec.write(CONSTRUCTOR.substitute(block_context))
 
+        if block['type'] == 'SectionSummary':
+            process_section_summary(schema_data, page_spec)
+        else:
```
Comment (jonnyshaw89) [inline tests/functional/generate_pages.py]: ??? Why ???
Is this just a copy / paste error?

File: tests/functional/pages/surveys/section_summary/property-details-summary.page.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+class PropertyDetailsSummaryPage extends QuestionPage {
+
+  constructor() {
+    super('property-details-summary');
+  }
+
+  insuranceTypeAnswer() { return '#insurance-type-answer-answer'; }
+
+  insuranceTypeAnswerEdit() { return '[data-qa="insurance-type-answer-edit"]'; }
```
Comment (jonnyshaw89) [inline tests/functional/pages/surveys/section_summary/property-details-summary.page.js:12]: Why do some section summary pages have edit and some don't?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: tests/functional/pages/surveys/section_summary/property-details-summary.page.js (hunk 10 above)
Comment (MIslam92) [reply tests/functional/pages/surveys/section_summary/property-details-summary.page.js:12]: Previously, It repeated a lot of the id's. on generation due to some code error in `generate_pages.py`, now that should be fixed.

File: tests/functional/generate_pages.py [hunk 11]
```diff
@@ -305,11 +305,6 @@ def process_block(block, dir_out, schema_data, spec_file, relative_require='..')
 
         if block['type'] == 'SectionSummary':
             process_section_summary(schema_data, page_spec)
-        else:
-            num_questions = len(block.get('questions', []))
-
-            for question in block.get('questions', []):
-                process_question(question, page_spec, num_questions, page_name)
 
         if block['type'] == 'Summary':
```
Comment (jonnyshaw89) [inline tests/functional/generate_pages.py]: `elif` so it doesn't need to evaluate this

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: tests/functional/generate_pages.py [hunk 12]
```diff
@@ -281,6 +288,9 @@ def process_block(block, dir_out, schema_data, spec_file, relative_require='..')
         page_spec.write(CLASS_NAME.substitute(block_context))
         page_spec.write(CONSTRUCTOR.substitute(block_context))
 
+        if block['type'] == 'SectionSummary':
+            process_summary(schema_data, page_spec)
+
         if block['type'] == 'Summary':
```
Comment (dwyeradam) [inline tests/functional/generate_pages.py]: These are doing the same thing so should be in a single if statement.

File: app/templates/sectionsummary.html [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+
+{% block main -%}
+
+<div>
+    <h1 class="saturn"></h1>
+    {% for group in content.summary.groups %}
+        <h2 class="summary__title saturn" id="{{group.id}}">{{group.title}}</h2>
+    {% endfor %}
+    <div class="print__message panel panel--simple panel--error">
+        <h2 class="saturn">Please remember to submit these answers.</h2>
```
Comment (dwyeradam) [inline app/templates/sectionsummary.html]: Should this be in section summary?

**Review by dwyeradam — CHANGES_REQUESTED** [review]

File: app/questionnaire/questionnaire_schema.py [hunk 14]
```diff
@@ -131,15 +131,15 @@ def get_questions_for_block(block_json):
...
+        self._sections_by_id = self.get_sections_by_id()
         self._groups_by_id = get_nested_schema_objects(self._sections_by_id, 'groups')
         self._blocks_by_id = get_nested_schema_objects(self._groups_by_id, 'blocks')
         self._questions_by_id = get_nested_schema_objects(self._blocks_by_id, 'questions')
         self._answers_by_id = get_nested_schema_objects(self._questions_by_id, 'answers')
         self.error_messages = self._get_error_messages()
         self.aliases = self._get_aliases()
 
-    def _get_sections_by_id(self):
+    def get_sections_by_id(self):
```
Comment (jonnyshaw89) [inline app/questionnaire/questionnaire_schema.py]: Why is this changed?

File: app/templates/partials/summary/summary.html [hunk 15]
```diff
@@ -4,10 +4,11 @@
   <div class="summary">
 
     {% for group in content.summary.groups %}
+      {% if group.title and content.summary.summary_type == 'summary' %}
```
Comment (jonnyshaw89) [inline app/templates/partials/summary/summary.html]: Do we not want to show the group titles in section summaries?

File: app/templating/view_context.py [hunk 16]
```diff
@@ -85,17 +86,30 @@ def build_view_context_for_question(current_location, variables, rendered_block,
     return context
 
 
-def build_view_context_for_summary(metadata, answer_store, schema_context, variables, csrf_token):
-    rendered_schema_json = renderer.render(g.schema.json, **schema_context)
-    summary_rendered_context = build_summary_rendering_context(g.schema, rendered_schema_json['sections'],
-                                                               answer_store, metadata)
+def build_view_context_for_summary(metadata, answer_store, schema_context, block_type, variables, csrf_token,
+                                   current_location):
+    sections = renderer.render(g.schema.json, **schema_context)['sections']
```
Comment (jonnyshaw89) [inline app/templating/view_context.py]: Move this inside `if block_type == 'Summary':`
so we don't render all the sections for section summaries

File: app/templating/view_context.py [hunk 17]
```diff
@@ -85,17 +86,30 @@ def build_view_context_for_question(current_location, variables, rendered_block,
...
+                summary_rendered_context = build_summary_rendering_context(g.schema, section, answer_store, metadata)
+                break
 
     context = {
         'csrf_token': csrf_token,
         'summary': {
             'groups': summary_rendered_context,
             'answers_are_editable': True,
             'is_view_submission_response_enabled': is_view_submitted_response_enabled(g.schema.json),
+            'summary_type': current_section,
```
Comment (jonnyshaw89) [inline app/templating/view_context.py]: `summary_type` is a `block_id`?? it looks like you are relying on the `block_id` being a specific value?
should this be `'summary_type': block_type`??

File: app/templating/view_context.py [hunk 18]
```diff
@@ -85,17 +86,30 @@ def build_view_context_for_question(current_location, variables, rendered_block,
...
+    current_section = current_location.block_id
+
+    if block_type == 'Summary':
+        summary_rendered_context = build_summary_rendering_context(g.schema, sections,
+                                                                   answer_store, metadata)
+    elif block_type == 'SectionSummary':
+        for section in sections:
+            group_ids = {group['id'] for group in section['groups']}
+            if current_location.group_id in group_ids:
+                section = [section]
```
Comment (jonnyshaw89) [inline app/templating/view_context.py]: change this to `sections = renderer.render([section], **schema_context)['sections']`
I think that should work anyway.

File: app/views/questionnaire.py [hunk 19]
```diff
@@ -672,7 +672,8 @@ def _get_front_end_navigation(answer_store, current_location, metadata, routing_
     completed_blocks = get_completed_blocks(current_user)
     navigation = Navigation(g.schema, answer_store, metadata, completed_blocks, routing_path)
     block_json = g.schema.get_block(current_location.block_id)
-    if block_json is not None and block_json['type'] in ('Question', 'Interstitial', 'Confirmation', 'Summary'):
+    if block_json['type'] in ('Question', 'Interstitial', 'Confirmation', 'SectionSummary', 'Summary') and \
```
Comment (jonnyshaw89) [inline app/views/questionnaire.py]: Just change this to `if block_json['type'] != 'Introduction' and block_json is not None:`

Comment (jonnyshaw89) [reply app/views/questionnaire.py]: or even try `if block_json['type'] != 'Introduction':` as i'm not sure why `block_json` would ever be `None`

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/templates/partials/summary/summary.html (hunk 15 above)
Comment (MIslam92) [reply app/templates/partials/summary/summary.html]: Based on the design it should only show the section titles on the section summaries
https://trello-attachments.s3.amazonaws.com/56d9b1dcc08c137b23e86234/5a69ed2da9df0bb13799f868/1038x671/c13873dbe5bba999ab052ceab568caea/Screen_Shot_2018-01-22_at_14.26.25.png

File: app/questionnaire/questionnaire_schema.py (hunk 14 above)
Comment (MIslam92) [reply app/questionnaire/questionnaire_schema.py]: Accidental change, changed it back

File: app/templates/partials/summary/summary.html (hunk 15 above)
Comment (jonnyshaw89) [reply app/templates/partials/summary/summary.html]: I mean, that images doesn't explicitly say that, it could just be that if there is only 1 group in a section then don't show the title?
Confirm this behaviour with Will

File: app/templating/view_context.py [hunk 20]
```diff
@@ -85,17 +86,30 @@ def build_view_context_for_question(current_location, variables, rendered_block,
...
+
+    if block_type == 'Summary':
+        sections = renderer.render(g.schema.json, **schema_context)['sections']
+        summary_rendered_context = build_summary_rendering_context(g.schema, sections,
+                                                                   answer_store, metadata)
+    elif block_type == 'SectionSummary':
+        for section in g.schema.json['sections']:
+            group_ids = {group['id'] for group in section['groups']}
+            if current_location.group_id in group_ids:
+                sections = [section]
```
Comment (jonnyshaw89) [inline app/templating/view_context.py]: I thought you rendered this?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/templating/view_context.py [hunk 21]
```diff
@@ -85,17 +86,30 @@ def build_view_context_for_question(current_location, variables, rendered_block,
...
+                                   current_location):
+
+    summary_rendered_context = None
+
+    if block_type == 'Summary':
+        sections = renderer.render(g.schema.json, **schema_context)['sections']
+        summary_rendered_context = build_summary_rendering_context(g.schema, sections,
+                                                                   answer_store, metadata)
+    elif block_type == 'SectionSummary':
+        for section in g.schema.json['sections']:
```
Comment (jonnyshaw89) [inline app/templating/view_context.py]: There is also no test coverage of this.

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/templating/view_context.py (hunk 20 above)
Comment (MIslam92) [reply app/templating/view_context.py]: I pass the section I'm on to the 'build_summary_rendering_context' which I thought builds and renders the page? So I just left it as that?

Comment (jonnyshaw89) [reply app/templating/view_context.py]: so why did you call `renderer.render` for `Summary` if you thought that?

Comment (MIslam92) [reply app/templating/view_context.py]: If I make it so instead it is `g.schema.json['sections']` for `Summary` it should work?

File: app/templating/view_context.py [hunk 22]
```diff
@@ -85,17 +86,30 @@ def build_view_context_for_question(current_location, variables, rendered_block,
...
+            group_ids = {group['id'] for group in section['groups']}
+            if current_location.group_id in group_ids:
+                sections = renderer.render(section, **schema_context)
```
Comment (jonnyshaw89) [inline app/templating/view_context.py]: maybe rename this to `rendered_section = `?
`sections` is wrong as its not a list

File: tests/functional/generate_pages.py [hunk 23]
```diff
@@ -280,8 +291,7 @@ def process_block(block, dir_out, schema_data, spec_file, relative_require='..')
         page_spec.write(HEADER.substitute(block_context))
         page_spec.write(CLASS_NAME.substitute(block_context))
         page_spec.write(CONSTRUCTOR.substitute(block_context))
-
-        if block['type'] == 'Summary':
+        if block['type'] == 'SectionSummary' or block['type'] == 'Summary':
```
Comment (jonnyshaw89) [inline tests/functional/generate_pages.py]: `if block['type'] in ('Summary', 'SectionSummary'):`

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/templates/sectionsummary.html [hunk 24]
```diff
@@ -0,0 +1,38 @@
...
+{% endblock %}
+
+{% block form_content %}
+{% endblock %}
+
+{% block main -%}
+
+<div>
+    {% for group in content.summary.groups %}
+        <h1 class="summary__title saturn" id="{{group.id}}">{{group.title}}</h1>
```
Comment (hamishtaplin) [inline app/templates/sectionsummary.html]: Is `.summary__title` still contained within a `.summary`?

File: data/en/test_section_summary.json [hunk 25]
```diff
@@ -0,0 +1,282 @@
...
+                        }]
+                    },
+                    {
+                        "type": "Question",
+                        "id": "repeating-block-2",
+                        "title": "{{answers.first_name[group_instance]}} {{answers.last_name[group_instance]}}",
+                        "questions": [{
+                            "id": "working-status-question",
+                            "title": "",
+                            "description": "What is the working status of this person?",
```
Comment (jonnyshaw89) [inline data/en/test_section_summary.json]: This is the title not the description. It make the summary show incorrectly otherwise

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/templates/partials/summary/summary.html [hunk 26]
```diff
@@ -4,10 +4,11 @@
   <div class="summary">
 
     {% for group in content.summary.groups %}
+      {% if group.title and content.summary.summary_type == 'Summary' %}
+        <h2 class="summary__title saturn" id="{{group.id}}">{{group.title}}</h2>
+      {% else %}
```
Comment (nealedj) [inline app/templates/partials/summary/summary.html]: Unnecessary `{% else %}`. This is empty and can be deleted.

File: app/templates/sectionsummary.html (hunk 24 above)
Comment (nealedj) [inline app/templates/sectionsummary.html]: If `group.title` is an empty string then this will render an empty `<h2>`. There's a check in `summary.html` to prevent this - could we add the check in here as well please?

File: app/templates/sectionsummary.html [hunk 27]
```diff
@@ -0,0 +1,38 @@
...
+  {% endif %}
+{% endblock %}
+
+{% block form_content %}
+{% endblock %}
+
+{% block main -%}
+
+<div>
+    {% for group in content.summary.groups %}
```
Comment (nealedj) [inline app/templates/sectionsummary.html]: This will render multiple titles if there are multiple groups. Presumably we should only ever have one group so the iterator is redundant. And adding multiple titles on top of each other would look odd any way.

Perhaps better to index directly? `{% set group = content.summary.groups | first  %}`

File: app/templating/view_context.py [hunk 28]
```diff
@@ -85,17 +86,30 @@ def build_view_context_for_question(current_location, variables, rendered_block,
...
+
+    summary_rendered_context = None
+
+    if block_type == 'Summary':
+        sections = renderer.render(g.schema.json, **schema_context)['sections']
+        summary_rendered_context = build_summary_rendering_context(g.schema, sections,
+                                                                   answer_store, metadata)
+    elif block_type == 'SectionSummary':
+        for section in g.schema.json['sections']:
+            group_ids = {group['id'] for group in section['groups']}
```
Comment (nealedj) [inline app/templating/view_context.py]: Unnecessary set comprehension. As `group_ids` in only used for checking if the current group is in it then this could be a generator:

```
group_ids = (group['id'] for group in section['groups'])
if current_location.group_id in group_ids:
    ...
```

File: app/views/questionnaire.py [hunk 29]
```diff
@@ -672,7 +672,7 @@ def _get_front_end_navigation(answer_store, current_location, metadata, routing_
     completed_blocks = get_completed_blocks(current_user)
     navigation = Navigation(g.schema, answer_store, metadata, completed_blocks, routing_path)
     block_json = g.schema.get_block(current_location.block_id)
-    if block_json is not None and block_json['type'] in ('Question', 'Interstitial', 'Confirmation', 'Summary'):
+    if block_json['type'] != 'Introduction':
```
Comment (nealedj) [inline app/views/questionnaire.py:681]: Nice 👍

**Review by nealedj — DISMISSED** [review]

File: app/templates/sectionsummary.html (hunk 27 above)
Comment (jonnyshaw89) [reply app/templates/sectionsummary.html]: Or should we have this title as the section title?

File: app/templates/section_confirmation.html [hunk 30]
```diff
@@ -0,0 +1,7 @@
+{% extends theme('layouts/_questionnaire.html') %}
+
+{% block page_title %}{{_("Submit answers")}} - {{survey_title}}{% endblock %}
+
+{% block section_submit_button %}
+  <button class="btn btn--primary btn--lg btn--loader js-btn-submit" data-qa="btn-submit" type="submit">Continue</button>
```
Comment (shahi645) [inline app/templates/section_confirmation.html]: @MIslam92 do we need `type="submit"`

File: app/templates/sectionsummary.html (hunk 27 above)
Comment (MIslam92) [reply app/templates/sectionsummary.html]: Well it is the "Section Summary" so makes sense to use the section title as the title?

File: app/templates/sectionsummary.html [hunk 31]
```diff
@@ -0,0 +1,41 @@
...
+{% endblock %}
+
+{% block form_content %}
+{% endblock %}
+
+{% block main -%}
+
+<div>
+    {% set group = content.summary.groups | first %}
+    {% for group in content.summary.groups %}
```
Comment (jonnyshaw89) [inline app/templates/sectionsummary.html]: Im not sure what these 2 lines do?
you assign group in the 1st then you reassign it in the 2nd

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/templates/sectionsummary.html (hunk 24 above)
Comment (jonnyshaw89) [reply app/templates/sectionsummary.html]: @MIslam92 we now arn't checking that there is a title

File: app/templates/sectionsummary.html [hunk 32]
```diff
@@ -0,0 +1,37 @@
...
+{% endblock %}
+
+{% block form_content %}
+{% endblock %}
+
+{% block main -%}
+
+<div>
+    {% set group = content.summary.groups | first %}
+        <h1 class="summary__title saturn" id="{{group.id}}">{{group.title}}</h1>
```
Comment (jonnyshaw89) [inline app/templates/sectionsummary.html]: The indentation is incorrect here.
also, what happens if the group title is blank?
should this not be the section title?

File: app/templating/view_context.py [hunk 33]
```diff
@@ -86,17 +87,30 @@ def build_view_context_for_question(current_location, variables, rendered_block,
...
-    summary_rendered_context = build_summary_rendering_context(g.schema, rendered_schema_json['sections'],
-                                                               answer_store, metadata)
+def build_view_context_for_summary(metadata, answer_store, schema_context, block_type, variables, csrf_token,
+                                   current_location):
+
+    summary_rendered_context = None
+
+    if block_type == 'Summary':
+        sections = renderer.render(g.schema.json, **schema_context)['sections']
+        summary_rendered_context = build_summary_rendering_context(g.schema, sections,
```
Comment (dwyeradam) [inline app/templating/view_context.py]: A little petty but it's clearer to show that the `build_summary_rendering_context` is doing the same thing for both summary pages if we moved it out of the ifs and pass a common variable into it. Maybe `section_list`. The section_list would just be a single item for the SectionSummary.
It might also be worth renaming `summary_rendered_context` to `summary_rendering_context` as it's the context used for rendering the summary. Yes it took me a while to get my head around that.

File: data/en/test_section_summary.json [hunk 34]
```diff
@@ -0,0 +1,282 @@
...
+    "description": "A questionnaire to test section summaries",
+    "theme": "default",
+    "legal_basis": "Voluntary",
+    "navigation": {
+        "visible": true
+    },
+    "sections": [{
+            "id": "property-details-section",
+            "title": "Property Details",
+            "groups": [{
```
Comment (dwyeradam) [inline data/en/test_section_summary.json]: We should have at least one section with multiple groups in to see how it deals with that. Also to ensure the correct title is being used on the pages don't have the section and group titles being the same.

File: data/en/test_section_summary.json [hunk 35]
```diff
@@ -0,0 +1,282 @@
...
+                                    {
+                                        "label": "Unemployed",
+                                        "value": "Unemployed"
+                                    }
+                                ]
+                            }]
+                        }]
+                    }
+                ],
+                "id": "repeating-group",
```
Comment (dwyeradam) [inline data/en/test_section_summary.json]: I'd take out repeating groups as the summary page isn't currently designed to cope with these and would detach from the work you're doing.

**Review by dwyeradam — CHANGES_REQUESTED** [review]

File: app/templating/view_context.py [hunk 36]
```diff
@@ -86,17 +87,32 @@ def build_view_context_for_question(current_location, variables, rendered_block,
...
 
     context = {
         'csrf_token': csrf_token,
         'summary': {
-            'groups': summary_rendered_context,
+            'groups': summary_rendering_context,
             'answers_are_editable': True,
             'is_view_submission_response_enabled': is_view_submitted_response_enabled(g.schema.json),
+            'summary_type': block_type,
+            'section_title': section_title,
```
Comment (dwyeradam) [inline app/templating/view_context.py]: This should be populated for both summary types and take out any if conditions in the html. So for the main summary the title would be "Your answers", for SectionSummary it would be the section title. Maybe look at renaming to something more specific to this use. Possibly just title.

**Review by dwyeradam — CHANGES_REQUESTED** [review]

File: app/templates/partials/summary/summary.html [hunk 37]
```diff
@@ -4,10 +4,9 @@
   <div class="summary">
 
     {% for group in content.summary.groups %}
-
-      {% if group.title %}
-      <h2 class="summary__title saturn" id="{{group.id}}">{{group.title}}</h2>
-      {% endif %}
+      {% if group.title and content.summary.summary_type == 'Summary' %}
+        <h2 class="summary__title saturn" id="{{group.id}}">{{group.title}}</h2>
+        {% endif %}
```
Comment (dwyeradam) [inline app/templates/partials/summary/summary.html]: I think we should remove the additional logic around summary type here to ensure consistency across the summary pages.

File: app/templates/sectionsummary.html [hunk 38]
```diff
@@ -0,0 +1,37 @@
...
+{% block main -%}
+
+<div>
+    {% set group = content.summary.groups %}
+    <h1 class="summary__title saturn" id="{{group.id}}">{{content.summary.title}}</h1>
+    <div class="print__hidden panel panel--simple panel--success">
+        <strong>This section is now complete</strong>
+        <p>You can review your answers below</p>
+    </div>
+</div>
```
Comment (dwyeradam) [inline app/templates/sectionsummary.html]: We have a lot of duplication here with the summary.html. These should be merged and a little logic just to show these text differences.

**Review by dwyeradam — CHANGES_REQUESTED** [review]

File: app/templates/summary.html [hunk 39]
```diff
@@ -15,15 +14,26 @@
...
     <h2 class="saturn">Please remember to submit these answers.</h2>
   </div>
   <div class="print__hidden panel panel--simple panel--warn">
     <h2 class="neptune">Please check your answers carefully before submitting.</h2>
   </div>
 </div>
+{% else %}
+<div>
+    {% set group = content.summary.groups %}
+    <h1 class="summary__title saturn" id="{{group.id}}">{{content.summary.title}}</h1>
```
Comment (dwyeradam) [inline app/templates/summary.html]: Why is this different from `<h1 class="saturn">{{content.summary.title}}</h1>`?

**Review by dwyeradam — CHANGES_REQUESTED** [review]
@MIslam92 For some reason I can't get into your House Details Section. I think it's to do with your skip condition based on insurance-type-answer. It shouldn't be in Nav really if you can't get to it.

**Comment by shahi645** [comment]
@dwyeradam I had that the other day, but i think its because theres a question in the first section that affects that

**Comment by MIslam92** [comment]
@dwyeradam you can only get into that section if you answered ''buildings" in the first question

**Comment by dwyeradam** [comment]
@MIslam92 Essentially we have a situation where having a sectionSummary in a section with no valid groups breaks our Navigation. Does sectionSummary need to be in it's own Group?

File: app/templating/view_context.py [hunk 40]
```diff
@@ -36,6 +37,7 @@ def build_view_context_for_non_question(variables, csrf_token, rendered_block):
         'variables': variables,
         'block': rendered_block,
         'csrf_token': csrf_token,
+        'block_type': rendered_block['type'],
```
Comment (jonnyshaw89) [inline app/templating/view_context.py]: Can we not just get the type in the template?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]
Was this done for a reason? Everywhere its added also has `block`

File: app/templates/summary.html [hunk 41]
```diff
@@ -15,15 +14,26 @@
...
     <h2 class="saturn">Please remember to submit these answers.</h2>
   </div>
   <div class="print__hidden panel panel--simple panel--warn">
     <h2 class="neptune">Please check your answers carefully before submitting.</h2>
   </div>
 </div>
+{% else %}
+<div>
+    {% set group = content.summary.groups %}
+    <h1 class="saturn" id="{{group.id}}">{{content.summary.title}}</h1>
```
Comment (jonnyshaw89) [inline app/templates/summary.html]: the `id` here doesn't work as a list doesn't have a property of `.id`

File: app/templates/confirmation.html [hunk 42]
```diff
@@ -3,5 +3,10 @@
 {% block page_title %}{{_("Submit answers")}} - {{survey_title}}{% endblock %}
 
 {% block submit_button %}
+
+{% if content.block or content.summary.summary_type == 'Summary' %}
```
Comment (jonnyshaw89) [inline app/templates/confirmation.html]: This logic needs to be inverted as discussed

File: app/templating/view_context.py [hunk 43]
```diff
@@ -44,6 +45,7 @@ def build_view_context_for_question(current_location, variables, rendered_block,
     context = {
         'variables': variables,
         'block': rendered_block,
+        'block_type': rendered_block['type'],
```
Comment (jonnyshaw89) [inline app/templating/view_context.py]: we can get rid of the `block_type`

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/templates/summary.html (hunk 41 above)
Comment (jonnyshaw89) [reply app/templates/summary.html]: why have you added this? it didn't have one before?

File: tests/functional/spec/summary_screen.spec.js [hunk 44]
```diff
@@ -7,7 +7,7 @@ const SummaryPage = require('../pages/surveys/summary_screen/summary.page.js');
 
 describe('Summary Screen', function() {
 
-  it('Given a survey has been completed when a summary page is displayed then it should contain all answers', function() {
+  it('Given a survey has been completed when a summary page is displayed then it should contain all answers @watch', function() {
```
Comment (dwyeradam) [inline tests/functional/spec/summary_screen.spec.js]: Remove this watch. Otherwise looking good.

**Review by dwyeradam — CHANGES_REQUESTED** [review]

File: tests/functional/spec/summary_screen.spec.js (hunk 44 above)
Comment (jonnyshaw89) [inline tests/functional/spec/summary_screen.spec.js]: remove the `@watch`

**Review by jonnyshaw89 — APPROVED** [review]

**Review by dwyeradam — APPROVED** [review]
